chpl/chpl-auth/src/main/java/gov/healthit/chpl/auth/authentication/UserAuthenticator.java

```diff
@@ -346,4 +339,55 @@ public void setJwtAuthor(final JWTAuthor jwtAuthor) {
         this.jwtAuthor = jwtAuthor;
     }
 
+    @Override
+    public String impersonateUser(final String username) throws UserRetrievalException,
+    JWTCreationException, UserManagementException {
+        JWTAuthenticatedUser user = (JWTAuthenticatedUser) Util.getCurrentUser();
+        if (user.getImpersonatingUser() != null) {
+            throw new UserManagementException("Unable to impersonate user while already impersonating");
+        }
+        UserDTO impersonatingUser = getUserByName(user.getSubjectName());
+        UserDTO impersonatedUser = getUserByName(username);
+        if (canImpersonate(user, impersonatedUser)) {
+            impersonatedUser.setImpersonatedBy(impersonatingUser);
+            return getJWT(impersonatedUser);
+        } else {
+            throw new UserManagementException("Unable to impersonate user");
+        }
+    }
+
+    @Override
+    public String unimpersonateUser(final User user) throws JWTCreationException, UserRetrievalException {
+        return getJWT(getUserByName(user.getSubjectName()));
+    }
+
+    private Boolean canImpersonate(final User actor, final UserDTO target) { //UserDTOs don't have the permissions. Ugh
+        for (GrantedPermission a : actor.getPermissions()) {
+            for (UserPermissionDTO t : securedUserManager.getGrantedPermissionsForUser(target)) {
+                if (!isGreaterRole(a.getAuthority(), t.getName())) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Compare acting user's role with targeted user's role.
+     * ROLE_ADMIN > all other roles
+     * ROLE_ADMIN is not > ROLE_ADMIN
+     * ROLE_ONC > all roles except ROLE_ADMIN and ROLE_ONC
+     * @param a acting user's ROLE
+     * @param t targeted user's ROLE
+     * @return true iff a > t
+     */
+    private Boolean isGreaterRole(final String a, final String t) {
```

Comment:
Could you make these variable names more descriptive? Maybe actingUserRole and targetedUserRole or something?

Reply:
Good news. This entire set of methods were removed with the transition to the new security thing